receiver/postgresqlreceiver/client.go

```diff
@@ -735,3 +742,91 @@ func tableKey(database, schema, table string) tableIdentifier {
 func indexKey(database, schema, table, index string) indexIdentifer {
 	return indexIdentifer(fmt.Sprintf("%s|%s|%s|%s", database, schema, table, index))
 }
+
+//go:embed templates/querySampleTemplate.tmpl
+var querySampleTemplate string
+
+func (c *postgreSQLClient) getQuerySamples(ctx context.Context, limit int64, logger *zap.Logger) ([]map[string]any, error) {
+	tmpl := template.Must(template.New("querySample").Option("missingkey=error").Parse(querySampleTemplate))
```

Comment:
I think you already used NamedArg in sqlserverreceiver, is there any reason that we don't use that?

Reply:
postgresql does not support named args https://github.com/lib/pq/issues/534